napari/_vispy/canvas.py

```diff
@@ -332,8 +332,12 @@ def _map_canvas2world(
             of the viewer.
         """
         nd = self.viewer.dims.ndisplay
-        transform = self.view.scene.transform
-        mapped_position = transform.imap(list(position))[:nd]
+        transform = self.view.camera._scene_transform
```

Comment:
what's the difference between `view.scene.transform` and `camera._scene_transform`? Shouldn't they be the same?

Reply:
I don't exactly know, here's my guess. view.scene.transform maps the coordinates from visual(data) to scene, camera._scene_transform transforms scene coordinates to camera coordinates and then project them to canvas.